src/Lib.P2P/SwarmService.cs

```diff
@@ -0,0 +1,1132 @@
+        ///   Is used to stop the task.  When cancelled, the <see cref="TaskCanceledException"/> is raised.
+        /// </param>
+        /// <returns>
+        ///   A task that represents the asynchronous operation.
+        /// </returns>
+        /// <remarks>
+        ///   If the peer is not conected, then nothing happens.
+        /// </remarks>
+        public Task DisconnectAsync(MultiAddress address, CancellationToken cancel = default)
+        {
+            Manager.Remove(address.PeerId);
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        ///   Start listening on the specified <see cref="MultiAddress"/>.
+        /// </summary>
+        /// <param name="address">
+        ///   Typically "/ip4/0.0.0.0/tcp/4001" or "/ip6/::/tcp/4001".
+        /// </param>
+        /// <returns>
+        ///   A task that represents the asynchronous operation.  The task's result
+        ///   is a <see cref="MultiAddress"/> than can be used by another peer
+        ///   to connect to tis peer.
+        /// </returns>
+        /// <exception cref="Exception">
+        ///   Already listening on <paramref name="address"/>.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   <paramref name="address"/> is missing a transport protocol (such as tcp or udp).
+        /// </exception>
+        /// <remarks>
+        ///   Allows other peers to <see cref="ConnectAsync(MultiAddress, CancellationToken)">connect</see>
+        ///   to the <paramref name="address"/>.
+        ///   <para>
+        ///   The <see cref="Peer.Addresses"/> of the <see cref="LocalPeer"/> are updated.  If the <paramref name="address"/> refers to
+        ///   any IP address ("/ip4/0.0.0.0" or "/ip6/::") then all network interfaces addresses
+        ///   are added.  If the port is zero (as in "/ip6/::/tcp/0"), then the peer addresses contains the actual port number
+        ///   that was assigned.
+        ///   </para>
+        /// </remarks>
+        public Task<MultiAddress> StartListeningAsync(MultiAddress address)
+        {
+            var cancel = new CancellationTokenSource();
+
+            if (!_listeners.TryAdd(address, cancel))
+            {
+                throw new Exception($"Already listening on '{address}'.");
+            }
+
+            // Start a listener for the transport
+            var didSomething = false;
+            foreach (var protocol in address.Protocols)
+                if (TransportRegistry.Transports.TryGetValue(protocol.Name, out var transport))
+                {
+                    address = transport().Listen(address, OnRemoteConnect, cancel.Token);
+                    _listeners.TryAdd(address, cancel);
+                    didSomething = true;
+                    break;
+                }
+
+            if (!didSomething)
+            {
+                throw new ArgumentException($"Missing a transport protocol name '{address}'.", nameof(address));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The parameter name '' is not declared in the argument list.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)